official/vision/beta/projects/mesh_rcnn/dataloaders/meshrcnn_input_test.py

```diff
@@ -0,0 +1,195 @@
+          gt_classes = labels['gt_classes']
+          gt_voxel = labels['gt_voxel']
+          gt_verts = labels['gt_verts']
+          gt_faces = labels['gt_faces']
+          gt_voxel_mask = labels['gt_voxel_mask']
+          gt_verts_mask = labels['gt_verts_mask']
+          gt_faces_mask = labels['gt_faces_mask']
+          rot_mat = labels['rot_mat']
+          trans_mat = labels['trans_mat']
+          trans_mat = labels['trans_mat']
+        else:
+          anchor_boxes = labels['anchor_boxes']
+          gt_boxes = labels['groundtruths']['boxes']
+          gt_classes = labels['groundtruths']['classes']
+          gt_voxel = labels['groundtruths']['voxel']
+          gt_verts = labels['groundtruths']['verts']
+          gt_faces = labels['groundtruths']['faces']
+          gt_voxel_mask = labels['groundtruths']['voxel_mask']
+          gt_verts_mask = labels['groundtruths']['verts_mask']
+          gt_faces_mask = labels['groundtruths']['faces_mask']
+          rot_mat = labels['rot_mat']
+          trans_mat = labels['trans_mat']
+          trans_mat = labels['trans_mat']
+
+        for j in range(params.parser.min_level, params.parser.max_level + 1):
+          self.assertAllEqual(
+              tf.shape(anchor_boxes[str(j)]),
+              [batch_size, 800 / (2 ** j), 800 / (2 ** j), 12])
+
+          if is_training:
+            self.assertAllEqual(
+                tf.shape(rpn_score_targets[str(j)]),
+                [batch_size, 800 / (2 ** j), 800 / (2 ** j), 3])
+
+            self.assertAllEqual(
+                tf.shape(rpn_box_targets[str(j)]),
+                [batch_size, 800 / (2 ** j), 800 / (2 ** j), 12])
+
+        self.assertNotEqual(
+            tf.reduce_max(gt_boxes), -1)
+
+        self.assertAllEqual(
+            tf.shape(gt_boxes),
+            [batch_size, params.parser.max_num_instances, 4])
+
+        self.assertAllEqual(
+            tf.shape(gt_classes),
+            [batch_size, params.parser.max_num_instances])
+
+        self.assertAllEqual(
+            tf.shape(gt_voxel),
+            [batch_size, params.parser.max_num_voxels, 3])
+
+        self.assertAllEqual(
+            tf.shape(gt_verts),
+            [batch_size, params.parser.max_num_verts, 3])
+
+        self.assertAllEqual(
+            tf.shape(gt_faces),
+            [batch_size, params.parser.max_num_faces, 3])
+
+        self.assertAllEqual(
+            tf.shape(gt_voxel_mask),
+            [batch_size, params.parser.max_num_voxels])
+
+        self.assertAllEqual(
+            tf.shape(gt_verts_mask),
+            [batch_size, params.parser.max_num_verts])
+
+        self.assertAllEqual(
+            tf.shape(gt_faces_mask),
+            [batch_size, params.parser.max_num_faces])
+
+        self.assertAllEqual(tf.shape(rot_mat), [batch_size, 3, 3])
+        self.assertAllEqual(tf.shape(trans_mat), [batch_size, 3])
+        self.assertAllEqual(tf.shape(trans_mat), [batch_size, 3])
+
+        if i >= 5:
```

Comment:
Possibly cleaner to change the for loop to use a slice of dataset? `for (image, labels) in dataset[0:6]` or so?

Reply:
The dataset object is not subscriptable.